crates/red_knot_python_semantic/resources/mdtest/protocols.md

```diff
@@ -1476,26 +1476,32 @@ signature implied by the `Callable` type is assignable to the signature of the `
 specified by the protocol:
 
 ```py
+from knot_extensions import TypeOf
+
 class Foo(Protocol):
     def __call__(self, x: int, /) -> str: ...
 
-# TODO: these fail because we don't yet understand that all `Callable` types have a `__call__` method,
-# and we therefore don't think that the `Callable` type is assignable to `Foo`. They should pass.
-static_assert(is_subtype_of(Callable[[int], str], Foo))  # error: [static-assert-error]
-static_assert(is_assignable_to(Callable[[int], str], Foo))  # error: [static-assert-error]
+static_assert(is_subtype_of(Callable[[int], str], Foo))
+static_assert(is_assignable_to(Callable[[int], str], Foo))
 
-static_assert(not is_subtype_of(Callable[[str], str], Foo))
-static_assert(not is_assignable_to(Callable[[str], str], Foo))
-static_assert(not is_subtype_of(Callable[[CallMeMaybe, int], str], Foo))
-static_assert(not is_assignable_to(Callable[[CallMeMaybe, int], str], Foo))
+# TODO: these should pass
+static_assert(not is_subtype_of(Callable[[str], str], Foo))  # error: [static-assert-error]
+static_assert(not is_assignable_to(Callable[[str], str], Foo))  # error: [static-assert-error]
+static_assert(not is_subtype_of(Callable[[CallMeMaybe, int], str], Foo))  # error: [static-assert-error]
+static_assert(not is_assignable_to(Callable[[CallMeMaybe, int], str], Foo))  # error: [static-assert-error]
```

Comment:
It's obviously fine to leave these as TODO for now -- but I'm not sure I understand where our implementation is lacking that makes them not pass yet?

Reply:
We still don't consider the types of protocol members at all when doing assignability or subtyping checks between any type `T` (a protocol type or otherwise) and a protocol type `P`. We still only check whether `T` _has_ the members specified by `P`; we do not yet check whether the types of the members on `T` are correct.

I'm working on it! Still not totally sure what the best design is when it comes to property members and method members, though. And want to add more tests first.